src/models/state-listener.ts

```diff
@@ -1,18 +1,19 @@
 import { State } from './state';
 
 /**
- * Represents a listener for a value in the state.
+ * Represents a listener for a value of type T in the state.
```

Comment:
Minor nit: This phrasing could be misinterpreted as "listens for all values of type T in the state"